### DISCUSSION TITLE
Hiding some Entries from specific Users

### DISCUSSION THREAD

**Discussion by fjahn** [statamic/cms#4836] [Q&A] [answered]
Unfortunately, I cannot re-mark the answer, but I did find a way to do this, see [my answer below](https://github.com/statamic/cms/discussions/4836?sort=old#discussioncomment-1727164).

Is there any way to _forcefully_ hide entries from users? I would like to show users only the entries they have created themself. I have successfully set up a [filter](https://statamic.dev/extending/query-scopes-and-filters#filters) that allows me to do just that, but users could still remove the filter and see entries created by other users.

**Comment by edalzell** [comment]
you can set permissions to allow users to only be able to access their own entries, but not hide from the listing.

**Reply by fjahn** [reply]
That's unfortunate, but thank you for the response! I already thought that this would be the case but wanted to make sure before giving up. Afterwards, I also found [this ancient feature request](https://github.com/statamic/ideas/issues/591) that describes the same problem.

Just to clarify: There are permissions for blocking users from *editing* other users' entries, but there is no way to completely hide them. In other words: There are separate permissions for `edit posts entries` and `edit other authors posts entries`, but there is only `view posts entries`, not `view other authors posts entries`.

I could only come up with one solution for my problem: Override the default `CollectionStore` with an implementation that only returns entries that the user is able to edit, but this is very bad for all sorts of reasons.

**Reply by velasco** [reply]
> you can set permissions to allow users to only be able to access their own entries, but not hide from the listing.

Hi @edalzell. Do you mean you can restrict users from seeing other users entries or can you just prevent them from changing other users entries?

**Comment by fjahn** [comment, marked as answer]
While planning to submit a PR to implement this, I did eventually find a way to do this using Laravel's service container!

You can extend Statamics `EntriesController` and override the `indexQuery` method. In there, you can add your own queries:

```php
use Statamic\Facades\User;
use Statamic\Http\Controllers\CP\Collections\EntriesController;

class FilteringAuthorEntriesController extends EntriesController
{
    protected function indexQuery($collection)
    {
        return parent::indexQuery($collection)
            ->where('author', User::current()->id());
    }
}

```

After that, you need to replace the original `EntriesController` with your own `FilteringAuthorEntriesController` in a service provider:

```php
public function register()
{
    $this->app->bind(EntriesController::class, FilteringAuthorEntriesController::class);
}
```

**Reply by edalzell** [reply]
very nice!
